### PR TITLE
Add support for percentages in column-gap.

### LINKED ISSUE #20498: Servo should support percentages in column-gap.
Gecko is gaining support for those, but Servo probably should too.

Code for this lives in `components/layout/multicol.rs`.

### PR DESCRIPTION
**Pull request by Ramshell**
# Changes in this PR
* I've removed the `if product == "gecko"` line in order to make the column_gap type "NonNegativeLengthOrPercentageOrNormal" always.

* Then I've changed this line
```rust
Either::First(len) => len.into(),
```
to
```rust
Either::First(len) => len.0.to_pixel_length(content_inline_size).into(),
```

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach build-geckolib` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #20498  (github issue number if applicable).

- [X] These changes do not require tests because there are tests already for this in wpt, in particular css/css-align/gaps/column-gap-parsing-001.html for the parsing bits, and css/css-multicol/multicol-gap-percentage-001.html for the layout bits.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20573)

### CODE REVIEW

**Comment by Ramshell** [comment]
I've run ./mach build -d; ./mach build-geckolib and ./mach test-tidy.

File: components/style/properties/longhand/column.mako.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ ${helpers.predefined_type(
 
 <%
 # FIXME(#20498): Servo should support percentages in column-gap.
-col_gap_type = "NonNegativeLengthOrPercentageOrNormal" if product == "gecko" else "NonNegativeLengthOrNormal"
+col_gap_type = "NonNegativeLengthOrPercentageOrNormal"
```
[the remark this answers is not in the document]
Comment (Ramshell) [reply components/style/properties/longhand/column.mako.rs]: Done :). Unless I didn't understood well what you asked :P

File: components/style/properties/longhand/column.mako.rs [hunk 2]
```diff
@@ -31,8 +31,7 @@ ${helpers.predefined_type(
 
 
 <%
-# FIXME(#20498): Servo should support percentages in column-gap.
-col_gap_type = "NonNegativeLengthOrPercentageOrNormal" if product == "gecko" else "NonNegativeLengthOrNormal"
+col_gap_type = "NonNegativeLengthOrPercentageOrNormal"
```
Comment (mbrubeck) [inline components/style/properties/longhand/column.mako.rs]: You can get rid of this variable, and pass `"length::NonNegativeLengthOrPercentageOrNormal"` directly as the second argument to `predefined_type`, below.

Comment (Ramshell) [reply components/style/properties/longhand/column.mako.rs]: You mean something like this?

```rust
${helpers.predefined_type(
    "column-gap",
    "length::NonNegativeLengthOrPercentageOrNormal",
    "Either::Second(Normal)",
    extra_prefixes="moz",
    servo_pref="layout.columns.enabled",
    animation_value_type="NonNegativeLengthOrPercentageOrNormal",
    spec="https://drafts.csswg.org/css-multicol/#propdef-column-gap",
    servo_restyle_damage = "reflow",
)}
```

Comment (mbrubeck) [reply components/style/properties/longhand/column.mako.rs]: Yes.

Comment (Ramshell) [reply components/style/properties/longhand/column.mako.rs]: Done :+1:

**Comment by Ramshell** [comment]
It looks like the failures are something to do with the ci environment. But tell me anyway if you want me to change something.

**Comment by mbrubeck** [comment]
Filed #20576 for the tests.

**Comment by Ramshell** [comment]
should I add
```
[css-align]
    skip: false
```
then?

**Comment by mbrubeck** [comment]
@bors-servo retry

Enabling the tests can be done as a separate PR.  It will require running the tests locally and adding [test expectations](https://github.com/servo/servo/blob/master/tests/wpt/README.md).  If you'd like to work on that next, that would be great!

**Pull request merged**